article/article.Rmd

```diff
@@ -0,0 +1,322 @@
+As these dates may refer to several possible dates,
+the function "opens" these values to include all the possible dates implied.
+
+```{r expand, warning=FALSE}
+dates_expand <- as_messydate(c("2001-01-01", "2001-01", "2001-01-01..2001-02-01",
+                               "{2001-01-01,2001-02-01}", "2001-XX-01"))
+expand(dates_expand)
+```
+
+## Contract
+
+The `contract()` function operates as the opposite of `expand()`.
+It contracts a list of dates into the abbreviated annotation of \pkg{messydates}.
+
+```{r contract, warning=FALSE, message=FALSE}
+contract(expand(dates_expand))
+```
+
+## Coerce from messydates
+
+Coercion functions coerce objects of `mdate` class to common date classes
+such as `Date`, `POSIXct`, and `POSIXlt`.
+Since `mdate` objects can hold multiple individual dates,
+an additional function must be passed as an argument so
+that multiple dates are "resolved" into a single date.
+For example, one might wish to use the earliest possible date
+in a range, or set, of expanded dates (`min`), or the latest possible date (`max`),
+or some notion of a central tendency (`mean`, `median`, or `modal`),
+or even a `random` selection from amongst the candidate dates.
+
+```{r coerce, warning=FALSE, message=FALSE}
+tibble::tibble(min = as.Date(dates_expand, min),
+               max = as.Date(dates_expand, max),
+               median = as.Date(dates_expand, median),
+               mean = as.Date(dates_expand, mean),
+               modal = as.Date(dates_expand, modal),
+               random = as.Date(dates_expand, random))
+```
+
+## Additional functionality 
+
+Several other functions are also offered in the \pkg{messydates} package.
+
+For example, one can check various logical tests for messy date objects.
+`is_messydate()` tests whether the object inherits the `mdate` class.
+`is_intersecting()` tests whether there is any intersection between two messy dates. `
+is_element()` similarly tests whether a messy date can be found within
+a messy date range or set.
+`is_similar()` tests whether two dates contain similar components.
+`is_precise()` tests for whether date is precise.
+
+```{r logical, warning=FALSE}
+is_messydate(as_messydate("2001-01-01"))
+is_messydate(as.Date("2001-01-01"))
+is_intersecting(as_messydate("2001-01"), as_messydate("2001-01-01..2001-02-22"))
+is_intersecting(as_messydate("2001-01"), as_messydate("2001-02-01..2001-02-22"))
+is_element(as_messydate("2001-01-01"), as_messydate("2001-01"))
+is_element(as_messydate("2001-01-01"), as_messydate("2001-02"))
+is_similar(as_messydate("2001-06-02"), as_messydate("2001-02-06"))
+is_similar(as_messydate("2001-06-22"), as_messydate("2001-02-06"))
+is_precise(as_messydate("2001-06-02"))
+is_precise(as_messydate("2001-02"))
+```
+
+Additionally, one can perform intersection (`md_intersect()`) and union (`md_union()`) on,
+inter alia, messy date class objects.
```

Comment:
Should we explain here why these aren’t class methods?

Reply:
I am not sure we need to explain or disclose here to users (or that this question will show up) but now I do wonder if they should be class methods? We do have related methods for arithmetic operations.